src/Component/Tests/Unit/Filter/DateFilterTest.php

```diff
@@ -0,0 +1,237 @@
+
+        $dataSource->expects($this->once())->method('restrict')->with('EXPR');
+
+        $this->filter->apply(
+            $dataSource,
+            'checkoutCompletedAt',
+            [
+                'to' => [
+                    'date' => '2016-12-06',
+                    'time' => '08:00',
+                ],
+            ],
+            [],
+        );
+    }
+
+    public function testFiltersDateToInclusive(): void
+    {
+        $dataSource = $this->createMock(DataSourceInterface::class);
+        $expressionBuilder = $this->createMock(ExpressionBuilderInterface::class);
+
+        $dataSource->method('getExpressionBuilder')->willReturn($expressionBuilder);
+        $expressionBuilder->expects($this->once())
+            ->method('lessThanOrEqual')
+            ->with('checkoutCompletedAt', '2016-12-06 08:00')
+            ->willReturn('EXPR');
+
+        $dataSource->expects($this->once())->method('restrict')->with('EXPR');
+
+        $this->filter->apply(
+            $dataSource,
+            'checkoutCompletedAt',
+            [
+                'from' => [
+                    'date' => '',
+                    'time' => '',
+                ],
+                'to' => [
+                    'date' => '2016-12-06',
+                    'time' => '08:00',
+                ],
+            ],
+            ['inclusive_to' => true],
+        );
+    }
+
+    public function testFiltersDateToWithDefaultTime(): void
+    {
+        $dataSource = $this->createMock(DataSourceInterface::class);
+        $expressionBuilder = $this->createMock(ExpressionBuilderInterface::class);
+
+        $dataSource->method('getExpressionBuilder')->willReturn($expressionBuilder);
+        $expressionBuilder->expects($this->once())
+            ->method('lessThan')
+            ->with('checkoutCompletedAt', '2016-12-06 23:59')
+            ->willReturn('EXPR');
+
+        $dataSource->expects($this->once())->method('restrict')->with('EXPR');
+
+        $this->filter->apply(
+            $dataSource,
+            'checkoutCompletedAt',
+            [
+                'to' => [
+                    'date' => '2016-12-06',
+                    'time' => '',
+                ],
+            ],
+            [],
+        );
+    }
+
+    public function testFiltersDateFromTo(): void
+    {
+        $dataSource = $this->createMock(DataSourceInterface::class);
+        $expressionBuilder = $this->createMock(ExpressionBuilderInterface::class);
+
+        $dataSource->method('getExpressionBuilder')->willReturn($expressionBuilder);
+
+        $expressionBuilder->expects($this->exactly(2))
+            ->method($this->logicalOr('greaterThanOrEqual', 'lessThan'))
+            ->willReturnCallback(function (string $field, string $value) {
+                return $value === '2016-12-05 08:00' ? 'EXPR1' : 'EXPR2';
+            });
+
+        $dataSource->expects($this->exactly(2))
+            ->method('restrict')
+            ->withConsecutive(['EXPR1'], ['EXPR2']);
```

Comment:
The method `withConsecutive()` is deprecated in PHPUnit 9 and removed in PHPUnit 10. Once we bump the dependency one day, it will break. I recommend to adjust it now so we don't have to iterate over this migration twice.

Refs: 
https://github.com/Sylius/SyliusGridBundle/actions/runs/19514483254/job/55862541450?pr=401#step:8:79
https://tomasvotruba.com/blog/how-to-upgrade-deprecated-phpunit-with-consecutive